cub/test/catch2_test_block_run_length_decode.cu

```diff
@@ -234,6 +234,9 @@ public:
       // decoding
       uint32_t num_valid_items = decoded_size - decoded_window_offset;
       run_length_decode.RunLengthDecode(decoded_items, relative_offsets, decoded_window_offset);
+
+      cub::CTA_SYNC(); // Sync before reusing aliased temp storage
+
```

Comment:
Arent we using the same storage in different ways below?

Reply:
No, `temp_storage` is a union, but `temp_storage.decode` is a struct, so `temp_storage.decode.store_decoded_runs_storage` and `temp_storage.decode.store_relative_offsets` are not aliased.

Reply:
```cpp
  union TempStorage
  {
    typename BlockLoadRunItemT::TempStorage load_uniques_storage;
    typename BlockLoadRunLengthsT::TempStorage load_run_lengths_storage;
    ::cuda::std::_If<TEST_RUN_OFFSETS_, typename BlockRunOffsetScanT::TempStorage, cub::NullType>
      run_offsets_scan_storage;
    struct
    {
      typename BlockRunLengthDecodeT::TempStorage run_length_decode_storage;
      typename BlockStoreDecodedItemT::TempStorage store_decoded_runs_storage;
      typename BlockStoreRelativeOffsetT::TempStorage store_relative_offsets;
    } decode;
  };
  ```

Reply:
I updated the location of the sync + the comment to clarify why the sync is needed here. It just has to be done somewhere in the `while` loop to make sure all warps stay on the same iteration.